### PR TITLE
use native RTCPeerConnection promises in Chrome 51

### PR DESCRIPTION
**Pull request by fippo**
**Description**
does not attempt to shim promises in Chrome 51+

**Purpose**
make adapter superflous

### CODE REVIEW

**Comment by KaptenJansson** [comment]
cc @jan-ivar

**Comment by fippo** [comment]
getStats-promises are still shimmed since https://bugs.chromium.org/p/chromium/issues/detail?id=499126 didn't include that.

**Comment by fippo** [comment]
I did a git reset, cherry-picked jib's stuff, then reapplied my change. I think that should merge without problems after you merge his PR... other than confusing @googlebot of course :-)

**Comment by KaptenJansson** [comment]
Merged his change but a conflict arose, I can fix it but rather you do it since it's your PR ;)

File: src/js/chrome/chrome_shim.js [hunk 1]
```diff
@@ -183,46 +183,48 @@ var chromeShim = {
...
+        };
+      });
+
+      ['setLocalDescription', 'setRemoteDescription', 'addIceCandidate']
+          .forEach(function(method) {
+            var nativeMethod = webkitRTCPeerConnection.prototype[method];
+            webkitRTCPeerConnection.prototype[method] = function() {
+              var args = arguments;
+              var self = this;
+              args[0] = new ((method === 'addIceCandidate')?
```
Comment (KaptenJansson) [inline src/js/chrome/chrome_shim.js]: Hate being this picky but there should be a space between ( and ?. Strange that the linter does not know about this.

File: src/js/chrome/chrome_shim.js [hunk 2]
```diff
@@ -183,46 +183,48 @@ var chromeShim = {
...
-            var promise = new Promise(function(resolve, reject) {
-              nativeMethod.apply(self, [args[0], resolve, reject]);
+    // add promise support -- natively available in Chrome 51
+    if (browserDetails.version < 51) {
+      ['createOffer', 'createAnswer'].forEach(function(method) {
+        var nativeMethod = webkitRTCPeerConnection.prototype[method];
+        webkitRTCPeerConnection.prototype[method] = function() {
+          var self = this;
+          if (arguments.length < 1 || (arguments.length === 1 &&
+              typeof(arguments[0]) === 'object')) {
```
Comment (KaptenJansson) [inline src/js/chrome/chrome_shim.js]: typeof should [not](https://google.github.io/styleguide/javascriptguide.xml?showone=Parentheses#Parentheses) have parenthesis.

Comment (KaptenJansson) [inline src/js/chrome/chrome_shim.js]: Again, we should be able to lint this.

**Comment by KaptenJansson** [comment]
LGTM

**Pull request merged**